packages/chevrotain/src/scan/tokens.ts

```diff
@@ -129,37 +119,37 @@ export function singleAssignCategoriesToksMap(
   path: TokenType[],
   nextNode: TokenType
 ): void {
-  forEach(path, (pathNode) => {
+  path.forEach((pathNode) => {
     nextNode.categoryMatchesMap![pathNode.tokenTypeIdx!] = true
   })
 
-  forEach(nextNode.CATEGORIES, (nextCategory) => {
+  nextNode.CATEGORIES?.forEach((nextCategory) => {
     const newPath = path.concat(nextNode)
     // avoids infinite loops due to cyclic categories.
-    if (!includes(newPath, nextCategory)) {
+    if (newPath.indexOf(nextCategory) === -1) {
       singleAssignCategoriesToksMap(newPath, nextCategory)
     }
   })
 }
 
 export function hasShortKeyProperty(tokType: TokenType): boolean {
-  return has(tokType, "tokenTypeIdx")
+  return tokType !== null && tokType.hasOwnProperty("tokenTypeIdx")
```

Comment:
`tokType` shouldn't ever be null according to the type signature, but it still seems to be happening for some reason. Lodash's `has` handles null fine, but we cannot call `null.hasOwnProperty`, so I had to check for it to make the tests pass.